### PR TITLE
Update docker-compose so it can run properly now

### PR DESCRIPTION
**Pull request by namnx228**
### Background

Update the docker-compose it can run properly with Redis support.

### Changes

- Update the docker-compose file to reflect the new mounting point
- Update doc file to describe how to run docker-compose with redis.

### Documentation

The change is small and can be understood by this pull request description. 

### Test Plan

Follow the step I described in the doc file change.

### PR Quality Checklist
- [X] My pull request is atomic and focuses on a single change.
- [X] I have thoroughly tested my changes with multiple different prompts.
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly and comprehensively.
- [X] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by namnx228** [comment]
@Torantulino @BillSchumacher  Please take a look

**Comment by ppetermann** [comment]
partially duplicates #2923

File: docker-compose.yml [hunk 1]
```diff
@@ -10,9 +10,9 @@ services:
...
     volumes:
-      - "./autogpt:/home/appuser"
-      - ".env:/home/appuser/.env"
+      - "./auto_gpt_workspace:/home/appuser/auto_gpt_workspace"
     profiles: ["exclude-from-up"]
 
   redis:
-    image: "redis/redis-stack-server:latest"
+    image: redis/redis-stack-server:latest
+    container_name: redis-stack-server
```
Comment (ppetermann) [inline docker-compose.yml:18]: don't name containers if not necessary, naming containers can create conflicts with other containers of the same name

File: docs/configuration/memory.md [hunk 2]
```diff
@@ -47,6 +47,17 @@ You can specify the memory index for redis using the following:
 MEMORY_INDEX=<WHATEVER>
 ```
 
+#### Using docker-compose
+In addition to the above changes in environment variables, change this one:
+```sh
+REDIS_HOST=redis-stack-server
```
Comment (ppetermann) [inline docs/configuration/memory.md:53]: that would explain why you are naming the container. use the service name instead (redis), and docker will take care of the rest

**Review by ppetermann — CHANGES_REQUESTED** [review]

**Comment by Pwuts** [comment]
Closing as deprecated; this was already fixed in #1843

**Pull request closed without merging**